### PR TITLE
Add 'reader' name to all produced DataArrays

### PR DESCRIPTION
**Pull request by djhoese**
I've brought up this feature at least 3 times in the past couple years but always find a way around not adding it. Well now I need it in Polar2Grid because I have too many enhancements and other configuration files that all parameters match *except* what reader the data come from. Or, if I'm being more honest, it is easier to specify by reader than listing the name for every product.

This also adds the option to use `'reader'` when filtering out enhancements. Not something I need to use directly in Satpy right now, but I need it in Polar2Grid.

 - [x] Tests added

### CODE REVIEW

**Comment by djhoese** [comment]
Darn it CodeFactor. I fixed a ton of issues but it still just says I fixed 1. I hate you.

File: satpy/tests/test_writers.py [hunk 1]
```diff
@@ -260,7 +258,6 @@ class TestEnhancerUserConfigs(_BaseCustomEnhancementConfigTests):
 
         """,
         ENH_ENH_FN: """
-sensor_name: visir/test_sensor
```
Comment (mraspaud) [inline satpy/tests/test_writers.py:263]: Why remove the sensor info here? We have them in the yaml, right?

**Review by mraspaud — COMMENTED** [review]
Looks good, just curious about the test modifications

File: satpy/tests/test_writers.py (hunk 1 above)
Comment (djhoese) [reply satpy/tests/test_writers.py:263]: We don't have them in enhancement files. I don't know why I added it here in the first place. The composites use this but it has no effect here for enhancements.

**Pull request merged**